keep-ui/app/alerts/alert-table.tsx

```diff
@@ -217,14 +227,74 @@ export function AlertTable({
     }),
   ];
 
+  const extraPayloadKeys = Array.from(
+    new Set(
+      alerts
+        .map((alert) => {
+          const extraPayload = getExtraPayloadNoKnownKeys(alert);
+          return Object.keys(extraPayload.extraPayload);
```

Comment:
If you're having trouble with naming:

```suggestion
          const { extraPayload } = getExtraPayloadNoKnownKeys(alert);
          return Object.keys(extraPayload);
```